source/index.ts

```diff
@@ -267,9 +269,17 @@ class Conf<T extends Record<string, any> = Record<string, unknown>> implements I
 
 	/**
 	Delete all items.
+
+	This resets known items to their default values, if defined by the `defaults` or `schema` option.
 	*/
 	clear(): void {
+		const keys = Object.keys(this.store);
+
 		this.store = createPlainObject();
+
+		for (const key of keys) {
+			this.reset(key);
+		}
```

Comment:
I think it'd make sense to emit the change event here, what do you think?

this.events.emit('change');

Reply:
That is already done, since we set `this.store`: https://github.com/sindresorhus/conf/blob/43c644f6fb0cd96a9c2e1f49b2ee0780cd089b4c/source/index.ts#L344

Reply:
@sindresorhus this is actually causing a regressive bug in my program. While we set this.store, we do it before reseting the parameters, and we set it to a blank object—which disobeys the type contract. Subsequently it's emitting an onDidAnyChange event where the current value is undefined (which disobeys the type guarantees for the store, given defaults)

Reply:
this doesn't cause a static typing violation because of the use Partial in several places where I don't believe it makes sense. Ideally it'd be possible to create a Conf with strict type guarantee. The type of details should statically guaranteed to be the base type.

Reply:
I don't have the time to look into this right now, so it would be good if you could open an issue instead. Even better if you could also submit some failing tests.

Reply:
I created an [issue](https://github.com/sindresorhus/conf/issues/138). The type safety bug here is pretty important. The clear() bug is as well, given that is breaks the contract with the client regarding types.